src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/azure_service_client.rb

```diff
@@ -14,6 +14,26 @@ class AzureServiceClient < MsRest::ServiceClient
     # @return [String] api version of the Azure in string format.
     attr_accessor :api_version
 
+    # @return [Hash{String=>String}] default request headers for Azure requests.
+    attr_accessor :request_headers
```

Comment:
You should have access to `request_headers`. We should not redefine it or i am missing something